### PR TITLE
(fix)capture: don't update org-id-locations if file is unknown

### PR DESCRIPTION
**Pull request by lennonhill**
My attempt to address https://github.com/org-roam/org-roam/issues/2102 .
Feedback is welcome.

###### Motivation for this change

### CODE REVIEW

File: org-roam-capture.el [hunk 1]
```diff
@@ -715,7 +715,10 @@ the current value of `point'."
         (when (find-buffer-visiting new-file)
           (kill-buffer (find-buffer-visiting new-file)))
         (delete-file new-file))
-    (org-id-add-location (org-roam-capture--get :id) (org-roam-capture--get :new-file))
+    (when-let (file (or (org-roam-capture--get :new-file)
```
Comment (jethrokuan) [inline org-roam-capture.el:718]: you're missing at least a bracket here, so file is always `nil`, and same below.

Comment (lennonhill) [reply org-roam-capture.el:718]: `when-let ` and `if-let` use the same SPEC, which "As a special case, interprets a SPEC of the form `(SYMBOL SOMETHING)` like `((SYMBOL SOMETHING))`", according to the documentation if `if-let`.

**Comment by lennonhill** [comment]
I've switched to double parentheses. The other syntax should be functionally equivalent, but it mostly exists for backward compatibility, so I'm happy to adopt the modern one.

File: org-roam-capture.el [hunk 2]
```diff
@@ -715,7 +715,10 @@ the current value of `point'."
         (when (find-buffer-visiting new-file)
           (kill-buffer (find-buffer-visiting new-file)))
         (delete-file new-file))
-    (org-id-add-location (org-roam-capture--get :id) (org-roam-capture--get :new-file))
+    (when-let ((file (or (org-roam-capture--get :new-file)
```
Comment (jethrokuan) [inline org-roam-capture.el:718]: It looks like for even for the `file` options, `:buffer` populated correctly, so it may be best to add remove this line.

Comment (lennonhill) [reply org-roam-capture.el]: This is also what I've been experiencing during my brief experiments.

**Pull request merged**